test: add light client sync test for forced update before update timeout

bshastry opened the pull request:
This PR adds a test case `test_light_client_sync_no_force_update` that verifies light client force update behavior before timeout threshold is reached. The test:

- Creates a valid but partial update (missing finalized header)
- Advances state to just before UPDATE_TIMEOUT threshold
- Verifies force update does not occur prematurely

This complements existing timeout tests by explicitly covering the following negative case (first branch in process_light_client_store_force_update evaluates to False because current slot is at the timeout threshold) that was previously uncovered

```
def process_light_client_store_force_update(store: LightClientStore, current_slot: Slot) -> None:
    if (
        current_slot > store.finalized_header.beacon.slot + UPDATE_TIMEOUT
        and store.best_valid_update is not None
    ):
        # Forced best update when the update timeout has elapsed.
        # Because the apply logic waits for `finalized_header.beacon.slot` to indicate sync committee finality,
        # the `attested_header` may be treated as `finalized_header` in extended periods of non-finality
        # to guarantee progression into later sync committee periods according to `is_better_update`.
        if store.best_valid_update.finalized_header.beacon.slot <= store.finalized_header.beacon.slot:
            store.best_valid_update.finalized_header = store.best_valid_update.attested_header
        apply_light_client_update(store, store.best_valid_update)
        store.best_valid_update = None
```

bshastry commented:
Might I trouble @etan-status for a review again since the PR covers light client spec? 😅

etan-status commented on tests/core/pyspec/eth2spec/test/altair/light_client/test_sync.py:
```diff
@@ -381,6 +381,74 @@ def run_lc_sync_test_upgraded_store_with_legacy_data(spec, phases, state, fork):
+@with_light_client
+@spec_state_test_with_matching_config
+@with_presets([MINIMAL], reason="too slow")
+def test_light_client_sync_no_force_update(spec, state):
```
Would recommend moving this up by a function (above `run_lc_sync_test_upgraded_store_with_legacy_data`), so that the `test_..._with_legacy_data` wrappers are located next to the `run_...` definition.

etan-status reviewed (approved):
Thanks for adding the test to also cover the test of force update before timeout is reached.

PR generally looks good to me once the comments are addressed.  If you could upload the generated test zip I can cross-check against Nimbus implementation.

bshastry commented:
> PR generally looks good to me once the comments are addressed. If you could upload the generated test zip I can cross-check against Nimbus implementation.

Thank you! I have addressed your comments. Please find the zip of the generated test vectors for altair attached.
[light_client_sync_no_force_update_altair.zip](https://github.com/user-attachments/files/18780776/light_client_sync_no_force_update_altair.zip)

etan-status commented:
Yaml looks good in nimbus-eth2

```diff
+ Light client - Sync - minimal/altair/light_client/sync/pyspec_tests/light_client_sync_no_f OK
```
